### PR TITLE
[MM][CG] Support ViT full CUDA graph for Idefics3 and SmolVLM

### PR DESCRIPTION
**Pull request by CHIPMUNK-T0T**
## Purpose

This PR adds ViT encoder CUDA Graph support for Idefics3 / SmolVLM as part of #38175.

Idefics3 computes vision position IDs from the patch attention mask during the encoder forward pass. To make the encoder compatible with CUDA Graph capture, this PR precomputes the position IDs and passes them as fixed graph inputs.

Changes:
- Implement `SupportsEncoderCudaGraph` for `Idefics3ForConditionalGeneration`
- Add CUDA Graph capture/replay using `pixel_values`, `pixel_attention_mask`, and `position_ids`
- Expose position-ID computation from `Idefics2VisionEmbeddings`
- Add SmolVLM coverage to `test_vit_cudagraph.py`

I checked #38175 and open PRs and found no existing PR covering Idefics3 / SmolVLM encoder CUDA Graph support.

AI assistance was used for this PR.

## Test Plan

Targeted test:

```bash
.venv/bin/python -m pytest tests/models/multimodal/generation/test_vit_cudagraph.py::test_vit_cudagraph_image -q -k idefics3
```

Also ran:

* Ruff on the changed Python files
* Python syntax checks
* Diff whitespace check
* A small end-to-end latency benchmark with encoder CUDA Graph disabled/enabled

## Test Result

* Targeted CUDA Graph test: `1 passed, 13 deselected`
* Ruff: passed
* Python syntax checks: passed
* Diff whitespace check: passed

### Benchmark

Environment:

* GPU: NVIDIA GeForce RTX 4070 12GB
* Model: `HuggingFaceTB/SmolVLM-256M-Instruct`
* Input: 1 generated 512×512 image + fixed prompt
* Batch size: 1
* `max_tokens=1`
* Warmup: 5 runs
* Measured: 30 runs
* Prefix cache: disabled
* MM processor cache: disabled

| Mode                   |      Mean |    Median |       P99 |
| ---------------------- | --------: | --------: | --------: |
| Encoder CUDA Graph off | 264.37 ms | 263.15 ms | 275.63 ms |
| Encoder CUDA Graph on  | 262.89 ms | 261.53 ms | 274.53 ms |

### CODE REVIEW

**Comment by CHIPMUNK-T0T** [comment]
Hi @ywang96,

This PR has been ready for review.
It adds ViT encoder CUDA Graph support for Idefics3 / SmolVLM, along with tests and a small benchmark.

When you have time, I'd appreciate any feedback. Thank you!

**Comment by CHIPMUNK-T0T** [comment]
Hi @ywang96

I've rebased onto the latest `main` and resolved the merge conflicts — the PR is now up to date and mergeable. The encoder CUDA graph test passes after the rebase (`test_vit_cudagraph_image[idefics3]`, SmolVLM-256M). Would appreciate a review when you have time. Thanks!

**Comment by Isotr0py** [comment]
Can you provide some accuracy benchmark to avoid accuracy regression? You can run OCRBench or MMMU with `evalscope` or `vlmevalkit` etc.
https://github.com/modelscope/evalscope/blob/54813f5c00d7b3dd02ccbdc78d382c019036b178/docs/en/benchmarks/ocr_bench.md

**Comment by CHIPMUNK-T0T** [comment]
@Isotr0py 
Thank you for your comment.

I ran OCRBench with EvalScope on the current PR branch, comparing encoder CUDA Graph ON and OFF by toggling only `cudagraph_mm_encoder`.

```text
OFF: --compilation-config '{"cudagraph_mm_encoder":false}'
ON:  --compilation-config '{"cudagraph_mm_encoder":true}'
```

Environment:
- GPU: NVIDIA GeForce RTX 4070 12GB
- Model: `HuggingFaceTB/SmolVLM-256M-Instruct`
- Dataset: OCRBench, 1,000 samples
- Decoding: greedy (`temperature=0`, `seed=42`)
- Batch size: 1
- Prefix cache: disabled
- MM processor cache: disabled

| Mode | OCRBench accuracy | Difference |
| --- | ---: | ---: |
| Encoder CUDA Graph OFF | 0.368 | |
| Encoder CUDA Graph ON | 0.365 | -0.003 |

The difference is about 0.2 standard errors and is within statistical uncertainty. Raw outputs were exactly identical for 969/1000 samples (96.9%).

The CUDA Graph run reported `hits=1000, misses=0, hit_rate=100.0%`, confirming that all evaluated samples used encoder CUDA Graph replay.

Based on these results, I did not observe a measurable accuracy regression.

**Review by Isotr0py — APPROVED** [review]
